src/AppBundle/Controller/FieldOfStudyController.php

```diff
@@ -27,16 +34,27 @@ public function editAction(Request $request, FieldOfStudy $fieldOfStudy = null)
             $fieldOfStudy = new FieldOfStudy();
             $isEdit = false;
         } else {
-            // Check if user is trying to edit FOS from department other than his own
-            if ($fieldOfStudy->getDepartment() !== $this->getUser()->getFieldOfStudy()->getDepartment()) {
+            // Check if a non-admin user is trying to edit FOS from department other than his own
+            if (!$this->isGranted(Roles::ADMIN) && $fieldOfStudy->getDepartment() !==
+                $this->getUser()->getFieldOfStudy()->getDepartment()) {
                 throw new AccessDeniedException();
             }
         }
         $form = $this->createForm(FieldOfStudyType::class, $fieldOfStudy);
         $form->handleRequest($request);
 
+        $departmentId = $request->query->get("departmentId");
+        // Use department from request if admin is creating a new field_of_study with id,
+        // else use department from current user
+        if (!$this->isGranted(Roles::ADMIN) && !$isEdit && !is_null($departmentId)) {
+            $department = $this->getDoctrine()->getRepository(Department::class)->findDepartmentById($departmentId)[0];
```

Comment:
Something like this would make Sonarcloud happy! (note: The indent may be off)
```suggestion
            $department = $this->getDoctrine()
                ->getRepository(Department::class)
                ->findDepartmentById($departmentId)[0];
```